oeps/oep-0008.rst

```diff
@@ -0,0 +1,323 @@
+==================================
+OEP-8: Unit Testing Best Practices
+==================================
+
++---------------+-------------------------------------------+
+| OEP           | :doc:`OEP-8 </oeps/oep-0008>`             |
++===============+===========================================+
+| Title         | Unit Testing Best Practices               |
++---------------+-------------------------------------------+
+| Last Modified | 2016-09-09                                |
++---------------+-------------------------------------------+
+| Author        | Jesse Zoldak                              |
++---------------+-------------------------------------------+
+| Arbiter       |                                           |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+| Created       | 2016-09-09                                |
++---------------+-------------------------------------------+
+| `Resolution`  |                                           |
++---------------+-------------------------------------------+
+| `Replaces`    |                                           |
++---------------+-------------------------------------------+
+| `Replaced-By` |                                           |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+The edX ecosystem has grown from a handful of developers working on a few
+repos to many developers working on many repos. As such, questions arise as to
+tools, frameworks, and patterns to use in unit testing. This proposal is
+intended to codify best practices and to provide a reference for contributors
+when making choices.
+
+
+Motivation
+==========
+
+Having common unit testing best practices across the organization's repo will
+increase the productivity of developers by reducing the ramp-up cost when
+moving among repos. It will also increase the quality of the software by
+decreasing the cognitive load on developers who must call to mind the various
+tips and techniques while using the disparate tools and frameworks.
```

Comment:
edx-platform is often an exception to these best practices.  Should we mention it up front? Also, many repos have smaller discrepancies from these best practices.  I don't imagine we're expecting that all repos will be brought into compliance with these practices.  A statement one way or the other would be good.